content/posts/building-images-with-img.md

```diff
@@ -0,0 +1,255 @@
+```groovy
+pipeline {
+  agent {
+    kubernetes {
+      label "img"
+      yamlFile 'img-resources/imgPod.yaml'
+    }
+  }
+  stages {
+    stage('Build and Push') {
+      steps {
+        container('img') {
+          sh """
+            img build -t gcr.io/melgin/img-hello-world ./hello-world
+            /home/user/google-cloud-sdk/bin/gcloud auth print-access-token | img login -u oauth2accesstoken --password-stdin https://gcr.io
+            img push gcr.io/melgin/img-hello-world
+          """
+        }
+      }
+    }
+  }
+}
+```
+
+In this Pipeline script, we load our agent `Pod` definition from `imgPod.yaml`, described above. Our actual steps are fairly straightforward and consist of three main actions within our `sh` step:
+
+1. We use `img build` to build our `Dockerfile` located in the `hello-world` subdirectory. This is a simple example that copies a "Hello World" script into the [Docker scratch image](https://hub.docker.com/_/scratch).
+2. We use `gcloud` to print our access token (for our Workload Identity account) and pipe that directly to `img login`, following the [GCR authentication documentation](https://cloud.google.com/container-registry/docs/advanced-authentication#access_token).
+3. Finally, we use `img push` to push our newly built image to our GCR repository.
+
+## Final Thoughts
+
+We've now successfully used `img` as a non-privileged, non-root Kubernetes `Pod` agent to build and push a container image from a Jenkins Pipeline. Before wrapping up, let's summarize some of the security implications of running `img` in this manner.
+
+First, `img` is running as a non-root user 1000 within a non-privileged container. This is an improvement from container building approaches that require root access (like `kaniko`) or using the `--privileged` flag (like Docker-in-Docker).
+
+However, there are security settings at the `PodSecurityPolicy`-level that require consideration, like setting `AppArmor` & `seccomp` policies to `unconfined` and allowing privilege escalation. While the default behavior in Kubernetes is to [set `seccomp` to `unconfined`](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#seccomp) and to [allow privilege escalation](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privilege-escalation), more security-conscious organizations might disallow these settings by default in their `PodSecurityPolicies`.
+
+Finally, without `procMount` set to `Unmasked`, the risk of child processes killing the parent `img` process remains. However, this risk is somewhat contained when `img` is run in an ephemeral container like it is as a Jenkins pod agent. (Note that this concern will be effectively resolved if or when `allowedProcMountTypes` is promoted from its current Alpha status.)
+
+For organizations particularly sensitive to avoiding running root or privileged workloads, `img` is a great candidate for building containers in Kubernetes.
```

Comment:
"running root or privileged workloads" - consider including container or something like that, for example:
"running root or privileged container workloads"